crates/core/tedge_watchdog/src/systemd_watchdog.rs

```diff
@@ -241,28 +258,63 @@ mod tests {
     async fn test_get_latest_health_status_message() -> Result<()> {
         let (mut sender, mut receiver) = mpsc::unbounded::<Message>();
         let health_topic = Topic::new("tedge/health/test-service").expect("Valid topic");
+        let base_timestamp = OffsetDateTime::now_utc();
+        let base_timestamp_str = base_timestamp
+            .format(&time::format_description::well_known::Rfc3339)
+            .unwrap();
+        for _x in 1..5i64 {
+            let timestamp = OffsetDateTime::now_utc();
+            let timestamp_str = timestamp
+                .format(&time::format_description::well_known::Rfc3339)
+                .unwrap();
 
-        for x in 1..5i64 {
             let health_status = json!({
                 "status": "up",
                 "pid": 123u32,
-                "time": x,
+                "time": timestamp_str,
             })
             .to_string();
             let health_message = Message::new(&health_topic, health_status);
             sender.publish(health_message).await?;
         }
 
-        let health_status = get_latest_health_status_message(3, &mut receiver).await;
-        assert_eq!(health_status.time, 3);
+        let health_status =
+            get_latest_health_status_message(base_timestamp_str.clone(), &mut receiver).await;
+
+        assert_eq!(health_status.unwrap().status, "up");
 
         let timeout_error = tokio::time::timeout(
             tokio::time::Duration::from_secs(1),
-            get_latest_health_status_message(5, &mut receiver),
+            get_latest_health_status_message(
+                "2050-06-22T17:03:14.000+02:00".to_string(),
+                &mut receiver,
+            ),
```

Comment:
```suggestion
            get_latest_health_status_message(
                base_timestamp + Duration::seconds(5)
                &mut receiver,
            ),
```
Just in case you wanted the old behaviour. This is also fine.